python/mxnet/io.py

```diff
@@ -6,6 +6,7 @@
 import ctypes
 import logging
 import threading
+import h5py
```

Comment:
we don't want to depend on h5py by default. Do it as an optional import. See cv2 dependency in recordio